src/Utils/kernel_launching.jl

```diff
@@ -43,10 +44,17 @@ function work_layout(grid, dims; include_boundaries=false)
                     (Int(√MAX_THREADS_PER_BLOCK), Int(√MAX_THREADS_PER_BLOCK))
 
     if include_boundaries
+        e = "include_boundaries kwarg requires the location kwarg to be specified"
+        isnothing(location) && throw(ArgumentError(e))
```

Comment:
if you use `instantiate(location)` it will throw an error, which might be sufficient error checking for this non-exported function

Reply:
Good point. Done.